lib/screens/v2/widget_instance/elevator_status.ex

```diff
@@ -0,0 +1,403 @@
+       ) do
+    stations = get_stations_from_entities(entities)
+
+    Alert.happening_now?(alert, now) &&
+      Enum.any?(stations, fn station ->
+        Enum.any?(stop_sequences, fn stop_sequence ->
+          station in stop_sequence and station != parent_station_id(t)
+        end)
+      end)
+  end
+
+  defp sort_elsewhere(e1, _e2, %__MODULE__{stop_sequences: stop_sequences}) do
+    stations = get_stations_from_entities(e1)
+
+    Enum.any?(stations, fn station ->
+      station in stop_sequences
+    end)
+  end
+
+  defp get_stations_from_entities(entities) do
+    entities
+    |> Enum.map(fn %{stop: stop_id} -> stop_id end)
+    |> Enum.filter(&String.starts_with?(&1, "place-"))
+  end
+
+  defp trim_and_page_alerts(pages) do
+    pages
+    |> Enum.flat_map(fn
+      %ListPage{} = page ->
+        split_list_page(page)
+
+      %DetailPage{} = page ->
+        [page]
+    end)
+    |> Enum.take(4)
+  end
+
+  defp split_list_page(%ListPage{stations: stations}) do
+    stations
+    |> Enum.chunk_every(@max_rows_per_page)
+    |> Enum.map(&%ListPage{stations: &1})
+  end
+
+  defp get_facility_by_id(entities, facilities) do
+    facility_in_entity =
+      entities
+      |> Enum.find_value(fn
+        %{facility: facility} -> facility
+        _ -> false
+      end)
+
+    facilities
+    |> Enum.find(fn %{id: id} -> id == facility_in_entity end)
+  end
+
+  defp serialize_closure(alert, %{name: name, id: id}, now) do
+    %{
+      elevator_name: name,
+      elevator_id: id,
+      timeframe: serialize_timeframe(alert, now),
+      description: alert.description
+    }
+  end
+
+  defp serialize_station(
+         {parent_station_id, alerts},
+         %__MODULE__{
+           facilities: facilities,
+           station_id_to_name: station_id_to_name,
+           station_id_to_icons: station_id_to_icons,
+           now: now
+         } = t
+       ) do
+    station_name = Map.fetch!(station_id_to_name, parent_station_id)
+
+    closures =
+      alerts
+      |> Enum.map(fn %Alert{
+                       informed_entities: entities
+                     } = alert ->
+        facility = get_facility_by_id(entities, facilities)
+
+        serialize_closure(alert, facility, now)
+      end)
+
+    %{
+      name: station_name,
+      icons: Map.fetch!(station_id_to_icons, parent_station_id),
+      elevator_closures: closures,
+      is_at_home_stop: parent_station_id == parent_station_id(t)
+    }
+  end
+
+  defp serialize_timeframe(%Alert{active_period: active_period} = alert, now) do
+    %{
+      happening_now: Alert.happening_now?(alert, now),
+      active_period: Enum.map(active_period, &Alert.ap_to_map/1)
+    }
+  end
+
+  defp alerts_by_station(alerts) do
+    alerts
+    |> Enum.map(fn %{informed_entities: informed_entities} = alert ->
+      {get_parent_station_id_from_informed_entities(informed_entities), alert}
+    end)
+    |> Enum.group_by(&elem(&1, 0), &elem(&1, 1))
```

Comment:
Could shorten this a bit since the values in the grouped lists are the same as the original list values! (third arg of `Enum.group_by/3` has a default value of `fn x -> x end`)
I vaguely remember writing this so apologies if I'm just cleaning up my own code here 😅
```suggestion
    |> Enum.group_by(&get_parent_station_id_from_informed_entities(&1.informed_entities))
```